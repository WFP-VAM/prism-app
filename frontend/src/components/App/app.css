html, body { height: 100% }
#root {
    height: 100%;
}

#app {
    display: flex;
    flex-direction: column;
    height: 100%;
}
