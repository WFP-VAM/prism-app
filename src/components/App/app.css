html, body { height: 100% }
#root {
    display: flex;
    flex-direction: column;
    height: 100%;
}

#app {
    flex: 1;
}
