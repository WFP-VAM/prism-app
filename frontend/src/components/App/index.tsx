import React from 'react';
import * as Sentry from '@sentry/browser';
import { useIsAuthenticated } from '@azure/msal-react';
import { ThemeProvider } from '@material-ui/core/styles';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Font } from '@react-pdf/renderer';
import { authRequired } from '../../config';
// Basic CSS Layout for the whole page
import './app.css';
import NavBar from '../NavBar';
import DataDrawer from '../DataDrawer';
import MapView from '../MapView';
import Login from '../Login';
import muiTheme from '../../muiTheme';
import Notifier from '../Notifier';
import AuthModal from '../AuthModal';

if (process.env.NODE_ENV && process.env.NODE_ENV !== 'development') {
  if (process.env.REACT_APP_SENTRY_URL) {
    Sentry.init({ dsn: process.env.REACT_APP_SENTRY_URL });
  } else {
    console.warn(
      'Sentry could not start. Make sure the REACT_APP_SENTRY_URL environment variable is set.',
    );
  }
}

// https://github.com/diegomura/react-pdf/issues/1991
Font.register({
  family: 'Roboto',
  fonts: [
    {
      src:
        'https://fonts.gstatic.com/s/roboto/v30/KFOmCnqEu92Fr1Me5WZLCzYlKw.ttf',
      fontWeight: 400,
    },
    {
      src:
        'https://fonts.gstatic.com/s/roboto/v30/KFOlCnqEu92Fr1MmEU9vAx05IsDqlA.ttf',
      fontWeight: 500,
    },
    {
      src:
        'https://fonts.gstatic.com/s/roboto/v30/KFOlCnqEu92Fr1MmWUlvAx05IsDqlA.ttf',
      fontWeight: 700,
    },
  ],
});

const Wrapper = () => {
  return (
    <div id="app">
      <NavBar />
      <Switch>
        <Route default>
          <MapView />
          <DataDrawer />
          <AuthModal />
        </Route>
      </Switch>
    </div>
  );
};

function App() {
  const isAuthenticated = useIsAuthenticated();

  return (
    <ThemeProvider theme={muiTheme}>
      {/* Used to show notifications from redux as a snackbar. Notifications are stored in notificationState */}
      <Notifier />
      <Router>
        {isAuthenticated || !authRequired ? <Wrapper /> : <Login />}
      </Router>
    </ThemeProvider>
  );
}

export default App;
