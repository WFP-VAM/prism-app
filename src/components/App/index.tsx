import React from 'react';
import * as Sentry from '@sentry/browser';
import { ThemeProvider } from '@material-ui/core/styles';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
// Basic CSS Layout for the whole page
import './app.css';
import NavBar from '../NavBar';
import DataDrawer from '../DataDrawer';
import MapView from '../MapView';
import NotFound from '../404Page';
import muiTheme from '../../muiTheme';
import Notifier from '../Notifier';

if (process.env.NODE_ENV && process.env.NODE_ENV !== 'development') {
  if (process.env.REACT_APP_SENTRY_URL) {
    Sentry.init({ dsn: process.env.REACT_APP_SENTRY_URL });
  } else {
    console.warn(
      'Sentry could not start. Make sure the REACT_APP_SENTRY_URL environment variable is set.',
    );
  }
}

function App() {
  return (
    <ThemeProvider theme={muiTheme}>
      {/* Used to show notifications from redux as a snackbar. Notifications are stored in notificationState */}
      <Notifier />
      <Router>
        <NavBar />
        <div id="app">
          <Switch>
            <Route exact path="/">
              <MapView />
              <DataDrawer />
            </Route>
            <Route default component={NotFound} />
          </Switch>
        </div>
      </Router>
    </ThemeProvider>
  );
}

export default App;
