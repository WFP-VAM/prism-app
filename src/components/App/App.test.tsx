import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

jest.mock('../NavBar/NavBar', () => 'mock-NavBar');

test('examples of some things', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
