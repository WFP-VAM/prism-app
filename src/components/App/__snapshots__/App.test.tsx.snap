// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<div>
  <mock-navbar />
  <div
    class="App"
  >
    <div
      style="outline: none;"
      tabindex="-1"
    >
      <div>
        hello
      </div>
    </div>
  </div>
</div>
`;
